[Elixir] Concept exercise: conditionals (#1129)
exercism/v3: languages/elixir/exercises/concept/conditionals/.docs/after.md

> You can use atoms whenever you have a set of constants you want to express. Using atoms opens a type-safe way of interacting with constant values. An atom is defined by its name, follwing the atom format:

Reviewer: ```suggestion
You can use atoms whenever you have a set of constants you want to express. Using atoms opens a type-safe way of interacting with constant values. An atom is defined by its name, following the atom format:
```

Reviewer: The `hints.md` file has a nice link to a document on atoms. Could you also link to that document somewhere in this document?

```elixir
# All atoms are preceeded with a ':' then follow with alphanumeric snake-cased characters
variable = :atom
```

> _Atoms_ are internally represented by an integer in a lookup table, which are set automatically.  It is not possible to change this internal value.  It is generally considered to be an [anti-pattern][anti-pattern] to dynamically create atoms from user supplied input.

Reviewer: This anti-pattern remark is a perfect example of something that you would indeed want to mention in the `after.md` document and not in the `introduction.md` document. Well done!

You can use [atoms][atom] whenever you have a set of constants you want to express. Using [atoms][atom] opens a type-safe way of interacting with constant values. An atom is defined by its name, following the [atom][atom] format:

```elixir
# All atoms are preceeded with a ':' then follow with alphanumeric snake-cased characters
variable = :an_atom
```

[_Atoms_][atom] are internally represented by an integer in a lookup table, which are set automatically.  It is not possible to change this internal value.  It is generally considered to be an [anti-pattern][anti-pattern] to dynamically create atoms from user supplied input.

> [_Atoms_][atom] are also often used to represent finite states. As many functions in Elixir's standard library return an atom to annotate the result:

Reviewer: ```suggestion
[_Atoms_][atom] are also often used to represent finite states. Many functions in Elixir's standard library return an atom to annotate the result:
```

```elixir
Enum.fetch([1], 0)
# => {:ok, 1}
Enum.fetch([1], 2)
# => :error
```

> When we want to have branching code, we can the use of the `cond/1` function:

Reviewer: ```suggestion
When we want to have branching code, we can use the `cond/1` function:
```

```elixir
cond do
  sometimes -> :this_might_be_the_way
  true -> :this_is_the_default_way
end
```

> At least one clause should evaluate to `true` or an error will be raised.

Reviewer: Should we perhaps make it clear that the error will be run-time? I don't know if that makes sense, so feel free to ignore.

```suggestion
At least one clause should evaluate to `true` or a run-time error will be raised.
```

Author: the dreaded runtime vs run-time!
